src/Stratis.Bitcoin.IntegrationTests/Miners/ProofOfStakeMintCoinsSteps.cs

```diff
@@ -0,0 +1,167 @@
+        }
+
+        private void it_mines_genesis_and_premine_blocks()
+        {
+            this.sharedSteps.MinePremineBlocks(this.nodes[PowMiner], PowWallet, PowWalletAccount, PowWalletPassword);
+        }
+
+        private void mine_coins_to_maturity()
+        {
+            this.nodes[PowMiner].GenerateStratisWithMiner(100);
+            this.sharedSteps.WaitForNodeToSync(this.nodes[PowMiner]);
+        }
+
+        private void a_proof_of_stake_node_with_wallet()
+        {
+            this.nodeGroupBuilder
+                    .CreateStratisPosNode(PosStaker)
+                    .MineCoinsFast()
+                    .Start()
+                    .NotInIBD()
+                    .WithWallet(PosWallet, PosWalletPassword)
+                    .Build();
+        }
+
+        private void it_syncs_with_proof_work_node()
+        {
+            this.posReceiverAddress = this.nodes[PosStaker].FullNode.WalletManager().GetUnusedAddress(new WalletAccountReference(PosWallet, PosWalletAccount));
+            var wallet = this.nodes[PosStaker].FullNode.WalletManager().GetWalletByName(PosWallet);
+            this.posReceiverPrivateKey = wallet.GetExtendedPrivateKeyForAddress(PosWalletPassword, this.posReceiverAddress).PrivateKey;
+
+            this.nodes[PosStaker].SetDummyMinerSecret(new BitcoinSecret(this.posReceiverPrivateKey, this.nodes[PosStaker].FullNode.Network));
+            this.nodeGroupBuilder.WithConnections().Connect(PowMiner, PosStaker);
+        }
+
+        private void sends_a_million_coins_from_pow_wallet_to_pos_wallet()
+        {
+            var context = SharedSteps.CreateTransactionBuildContext(
+                PowWallet,
+                PowWalletAccount,
+                PowWalletPassword,
+                this.posReceiverAddress.ScriptPubKey,
+                Money.COIN * 1000000,
+                FeeType.Medium,
+                101);
+
+            context.OverrideFeeRate = new FeeRate(Money.Satoshis(20000));
+
+            var unspent = this.nodes[PowMiner].FullNode.WalletManager().GetSpendableTransactionsInWallet(PowWallet);
+            var coins = new List<Coin>();
+
+            var blockTimestamp = unspent.OrderBy(u => u.Transaction.CreationTime).Select(ts => ts.Transaction.CreationTime).First();
+            var transaction = new Transaction
+            {
+                Time = (uint)blockTimestamp.ToUnixTimeSeconds()
+            };
+
+            foreach (var item in unspent.OrderByDescending(a => a.Transaction.Amount))
+            {
+                coins.Add(new Coin(item.Transaction.Id, (uint)item.Transaction.Index, item.Transaction.Amount, item.Transaction.ScriptPubKey));
+            }
+
+            var coin = coins.First();
+            var txIn = transaction.AddInput(new TxIn(coin.Outpoint, this.powSenderAddress.ScriptPubKey));
+            transaction.AddOutput(new TxOut(new Money(9699999999995400), this.powSenderAddress.ScriptPubKey));
```

Comment:
I am not too sure this would work.  The statement at line 116 is the same as just getting the first item in the list - unless this is not consistent when the list is being built.  Also the line could be moved closer to where it's used.

So something like this 

![image](https://user-images.githubusercontent.com/16000683/38772657-19e09a58-4034-11e8-8ad9-d0ffed04dec1.png)